daml/Marketplace/Clearing/Market/Model.daml

```diff
@@ -0,0 +1,51 @@
+module Marketplace.Clearing.Market.Model where
+
+import DA.Finance.Types
+import DA.Next.Set (Set)
+
+template FairValueCalculationRequest
+  with
+    operator : Party
+    provider : Party
+    customer : Party
+    optListingIds : Optional [Text]
+    currency : Id
+    upTo : Time
+  where
+    signatory provider
```

Comment:
Should the `operator` also be a signature ?

Reply:
I can add this. I'm curious though, I've noticed on multiple models I've worked on that the `operator` is a `signatory` on the majority of contracts, even ones the `operator` does not need to perform any action on. What is the general purpose of this? Is it to provide visibility to the ledger administrator (and wouldn't this be a possible privacy concern?). Or maybe to disallow a party from archiving contracts without using a `choice`?

Reply:
Good question - this main use of this pattern is to enforce that contracts are created only via service `choice`s which require the signature from the `operator` of that Marketplace. In other words, a `provider` (or a `customer` or even an `operator`...) can't simply create contracts outside of the "approved" service context, they must follow the explicit workflow provided by a service.

Regarding the privacy concern of this pattern, it's correct that an `operator` will have visibility of all the contracts they are a signatory to. Whether this is an issue or not likely depends on how you view the the `operator` which runs such a Marketplace. Should this be viewed as `admin` access (akin to a database super user access) which can view and monitor an entire Marketplace (makes it easier to audit, regulatory reporting, etc.) ? Or, should templates only have parties which are strictly required be allowed ? I think there's a trade off to be made here and currently we're enforcing having the correct signatures over privacy.

@GeorgSchneider , do you have view on this ?

Reply:
yep, having the operator on there is merely to be able to restrict "valid" contracts in an operator-centric marketplace. That is, a given marketplace (operator) will only accept contracts that are signed by the operator, such that different parties cannot act "outside" of the provided service framework. In the decentralized operating model, the services would be "self-signed" with the providers as the respective operators. Re privacy, it is by design that an operator-centric marketplace would give full visibility for the operator. The decentralized mode is one that would retain certain privacy properties.

In summary I'd suggest to add the operator here as the signatory, assuming that the service this originates from does have the operator signature on it so it can be done easily.

Reply:
This makes a lot of sense, thanks all for the explanations. I actually noticed when adding the `operator` that I was required then to create these contracts in the trigger via the `Service`, so that gave me a nice practical example of why this pattern is useful 😃 .